spring-ws-core/src/main/java/org/springframework/ws/soap/saaj/SaajSoapMessageFactory.java

```diff
@@ -233,7 +235,19 @@ private MimeHeaders parseMimeHeaders(InputStream inputStream) throws IOException
 		MimeHeaders mimeHeaders = new MimeHeaders();
 		if (inputStream instanceof TransportInputStream) {
 			TransportInputStream transportInputStream = (TransportInputStream) inputStream;
-			for (Iterator<String> headerNames = transportInputStream.getHeaderNames(); headerNames.hasNext();) {
+			
+			//Remove duplicate keys
+			Iterator<String> headerNamesWithDuplicate = transportInputStream.getHeaderNames();
```

Comment:
transportInputStream.getHeaderNames() --> return list of header names (with duplicate keys)

So if we call with duplicate keys, transportInputStream.getHeaders(keyName) we have the same answer in duplicate...